src/Groups/matrices/stuff_field_gen.jl

```diff
@@ -0,0 +1,52 @@
+import Hecke: evaluate, field_extension, FinField, FinFieldElem, PolyElem
+
+export primitive_element
+
+function _change_type(f::PolyElem{T}) where T <: FinFieldElem
+   e,p = ispower(order(base_ring(f)))
+   F = GF(Int(p),Int(e))[1]
+   t = PolynomialRing(F,"t")[2]
+   return sum([t^i*F(lift(coeff(f,i))) for i in 0:degree(f)])
+end
+
+# return a generator for the unit group of F
+function _centralizer(f::PolyElem{T}) where T <: FinFieldElem
```

Comment:
But there is no `F` specified, so what does that comment refer to? Perhaps `R[X]/(f)` where `R = base_ring(f)` ?

Reply:
Yes, I added it.